app/services/rbac_api.rb

```diff
@@ -1,32 +1,63 @@
 # frozen_string_literal: true
 
-# This class is meant to handle calls to the RBAC light API
+require 'insights-rbac-api-client'
+
+# This class is meant to handle calls to the RBAC API
 class RbacApi
+  COMPLIANCE_FULL_ACCESS = 'compliance:*:*'
+  APPS = 'compliance,inventory'
+  ANY = '*'
+
   def initialize(b64_identity)
-    @url = URI.parse("#{URI.parse(Settings.rbac_url)}"\
-                     "#{Settings.path_prefix}/rbac/v1/access/")
+    @client = RBACApiClient::AccessApi.new
```

Comment:
I'm waging if we should have this service a instantiable or to convert it to a singleton. As far as I understand, this `AccessApi` does not hold information that can leak into other contexts, i.e. we could theoretically share it among all requests:

```
class Rbac
  # ...
  API_CLIENT = RBACApiClient::AccessApi.new

  class << self
    def load_user_permissions(identity)
      # ...
    end

    private

    # ...
  end
end

```

Reply:
I guess the only downside would be to pass the identity to the load method, but as it would be called only once and from a single location (where the header actually lives) I think it is fine, wdyt?

Reply:
Sounds reasonable to me, will try it :+1:

Reply:
The point is to have the singleton class as a glue between our controller(s) and the API client.